stripe/util.py

```diff
@@ -34,26 +35,32 @@
 logger: logging.Logger = logging.getLogger("stripe")
 
 __all__ = [
+    # We export 'io' from this module for monkey-patching in
+    # tests.
     "io",
-    "parse_qsl",
     "log_info",
     "log_debug",
     "dashboard_link",
     "logfmt",
-    "deprecated",
+    "parse_qsl",
+    "_log_info",
+    "_log_debug",
+    "_dashboard_link",
+    "_deprecated",
 ]
 
-if hasattr(typing_extensions, "deprecated"):
```

Comment:
This `hasattr` thing was actually interfering with IDE support, and it was too conservative. Even if you have an older version of `typing_extensions` without "deprecated"
* It won't crash, because `if TYPE_CHECKING`.
* It turns out that *typeshed* actually has a type definition for `typing_extensions.deprecated` which will be used even if your `typing_extensions` version is old. So as long as your typechecker version is relatively recent/includes a relatively recent version of typeshed, then you're good to go.